### PR TITLE
Add `Config` stdlib page

### PR DESCRIPTION
**Pull request by colinhacks**

### CODE REVIEW

File: docs/stdlib/cfg.rst [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+The ``cfg`` module contains a set of types and scalars used for configuring
+EdgeDB.
+
+
+.. list-table::
+  :class: funcoptable
+
+  * - **Type**
+    - **Description**
+  * - :eql:type:`cfg::AbstractConfig`
```
Comment (1st1) [inline docs/stdlib/cfg.rst:16]: Querying `AbstractConfig` currently raises an ISE (which is a bug, but still). I recommend just documenting `cfg::Config` which acts like a view -- it reflects whatever setting values are set in your instance/db/session maintaining the correct resolution order.

File: docs/stdlib/cfg.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    - A subclass of ``AuthMethod`` indicating an "always trust" policy (no
+      authentication).
+  * - :eql:type:`cfg::SCRAM`
+    - A subclass of ``AuthMethod`` indicating password-based authentication.
+
+
+----------
+
+
+.. eql:type:: cfg::AbstractConfig
```
Comment (1st1) [inline docs/stdlib/cfg.rst:33]:
```suggestion
.. eql:type:: cfg::Config
```

File: docs/stdlib/cfg.rst [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+  An abstract type representing the configuration of an instance or database.
+
+  Is extended by three non-abstract subclasses:
+  ``cfg::Config``, ``cfg::InstanceConfig``, and ``cfg::DatabaseConfig``. At
+  the moment, these subclasses do not declare any additional properties or
+  links beyond what they inherit from ``cfg::AbstractConfig``.
```
Comment (1st1) [inline docs/stdlib/cfg.rst:40]: I'd just drop this paragraph for now.

File: docs/stdlib/cfg.rst [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    * - ``listen_addresses``
+      - ``multi std::str``
+      - N/A
+    * - ``auth``
+      - ``multi cfg::Auth``
+      - N/A
+    * - ``allow_dml_in_functions``
+      - ``std::bool``
+      - N/A
+    * - ``shared_buffers``
```
Comment (1st1) [inline docs/stdlib/cfg.rst:66]: Can we add copy from guides/configuration to here?

File: docs/stdlib/cfg.rst [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+      define the set of configuruation settings supported by EdgeDB.
+  * - :eql:type:`cfg::Auth`
+    - An object type representing an authentication profile.
+  * - :eql:type:`cfg::AuthMethod`
+    - An abstract object type representing a method of authentication
+  * - :eql:type:`cfg::Trust`
+    - A subclass of ``AuthMethod`` indicating an "always trust" policy (no
+      authentication).
+  * - :eql:type:`cfg::SCRAM`
+    - A subclass of ``AuthMethod`` indicating password-based authentication.
```
Comment (1st1) [inline docs/stdlib/cfg.rst:27]: Add `cfg::memory`?

**Review by 1st1 — COMMENTED** [review]
I've left some comments

File: docs/stdlib/cfg.rst (hunk 5 above)
Comment (colinhacks) [reply docs/stdlib/cfg.rst:27]: Added

File: docs/stdlib/cfg.rst (hunk 4 above)
Comment (colinhacks) [reply docs/stdlib/cfg.rst:68]: I think that should probably stay as a guide (though it needs some work)

Comment (colinhacks) [reply docs/stdlib/cfg.rst:151]: Added a setting reference to this page

File: docs/stdlib/cfg.rst [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+Resource usage
+--------------
+
+:eql:synopsis:`effective_io_concurrency (int64)`
+    Sets the number of concurrent disk I/O operations that can be
+    executed simultaneously. Corresponds to the PostgreSQL
+    configuration parameter of the same name.
+
+:eql:synopsis:`query_work_mem (str)`
```
Comment (1st1) [inline docs/stdlib/cfg.rst:82]:
```suggestion
:eql:synopsis:`query_work_mem (cfg::memory)`
```

File: docs/stdlib/cfg.rst [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+:eql:synopsis:`effective_cache_size (str)`
+    Sets the planner's assumption about the effective size of the disk
+    cache that is available to a single query. Corresponds to the
+    PostgreSQL configuration parameter of the same name.
+
+
+Client connections
+------------------
+
+:eql:synopsis:`client_idle_timeout (int16)`
```
Comment (1st1) [inline docs/stdlib/cfg.rst:110]: This looks like content from old master branch; please copy this page from the latest master, where this setting is renamed, has slightly different docs, has different type (std::duration). Also there are a few new settings not mentioned here.

File: docs/stdlib/index.rst [hunk 8]
```diff
@@ -32,6 +32,7 @@ The standard library reference is grouped by data type.
     sequence
     sys
     uuid
+    cfg
```
Comment (1st1) [inline docs/stdlib/index.rst:35]: Can we push it to the very bottom?

**Review by 1st1 — APPROVED** [review]

**Pull request merged**